src/types/dependency-container.ts

```diff
@@ -5,8 +5,9 @@ import ValueProvider from "../providers/value-provider";
 import ClassProvider from "../providers/class-provider";
 import constructor from "./constructor";
 import RegistrationOptions from "./registration-options";
+import Disposable from "./disposable";
 
-export default interface DependencyContainer {
+export default interface DependencyContainer extends Disposable {
```

Comment:
`Disposable` is an interface - should this be `implements`?

Reply:
Interfaces `extend` other interfaces in TypeScript. Only classes can `implement`